### PR TITLE
Use grid common search action where possible (1.7.8 target)

### LINKED ISSUE #13448: [SF Migration] Harmonize controller grid search action
We have a generic and reusable [searchGridAction](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/CommonController.php#L335) in CommonController but some Controllers use their own `searchAction` rather than the reusable one.

For each Controller which has a custom searchAction, check if there is a valid need/reason for it, else make it use the generic `searchGridAction`.

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Implements CommonController::searchGridAction where it was possible to do easily. Custom search actions are only left for multi-grid pages. Deprecated unusable search actions
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes, some constructors have been modified (php classes and Symfony service definition) but it was only `final` classes
| Deprecations? | no
| Fixed ticket? | #13448
| How to test?  | Check that grid filtering works as before (if these changes broke something, then filtering would not work at all, not necessary to test all the edge-cases). Modified grids list below.<br>- categories,<br>- credit slips,<br>- customers,<br>- languages,<br>- currencies,<br>- taxes,<br>- log,<br>- meta,<br>- requestsql,<br>- webserviceKey

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18123)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by zuk3975** [comment]
@jolelievre mentioned that some bugs have occurred in older grids due to missing grid_ids when merged similar pr to 1.7.7.x (https://github.com/PrestaShop/PrestaShop/pull/18130). And some BC breaks might be introduced. This pr shares the same changes of CommonController (as in abovementioned pr) so we need his input before merging this one.

**Comment by jolelievre** [comment]
Yes indeed, there is a breaking change on old grids (using controller/action matching instead of filterId)
Here is the PR with fix proposal: https://github.com/PrestaShop/PrestaShop/pull/18228

**Review by jolelievre — CHANGES_REQUESTED** [review]
Hi @zuk3975 
Sorry but I block this PR only so we don't forget to update it, as we discussed there was a breaking change in CommonController which is fixed in this PR https://github.com/PrestaShop/PrestaShop/pull/18228

So a new interface `IdentifiableGridDefinitionFactoryInterface` has been introduced so that `CommonController` can know how to render the response. For easier integration an abstract class `AbstractIdentifiableGridDefinitionFactory` implementing this interface has been introduced So you just need to change the parent class of your grid definitions to make it all work fine

**Comment by jolelievre** [comment]
Blocked until https://github.com/PrestaShop/PrestaShop/pull/18228 is merged (and merged into develop)

**Comment by zuk3975** [comment]
> Sorry but I block this PR only so we don't forget to update it, as we discussed there was a breaking change in CommonController which is fixed in this PR #18228

Sure, no problem, its for develop, so no need to rush.

**Comment by khouloudbelguith** [comment]
Hi @zuk3975,

Can you please resolve conflicts.

Thanks!

File: src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php
Comment (jolelievre) [inline src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php:216]: I think you could clean the constructor (since the arguments are not used any more) and the service definition no?

It might be considered a BC break although it's arguable since this is a final class not destined to be extended nor constructed manually, so at worst it's a small one

File: src/Core/Grid/Definition/Factory/WebserviceKeyDefinitionFactory.php
Comment (jolelievre) [inline src/Core/Grid/Definition/Factory/WebserviceKeyDefinitionFactory.php:194]: Same as `LogGridDefinitionFactory` construction could be cleaned

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php:89]: Why modify this controller and not use the common search one instead? This could simply be deprecated and not modified right?

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php]: Same here, deprecation isn't enough?

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php
Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php:89]: By the way the deprecate comment is missing

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php]: Missing `@deprecated` comment, and modification shouldn't be required

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MetaController.php]: Modification not required

File: src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php]: Modification not required

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/categories.yml
Comment (jolelievre) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/categories.yml]: Changing the path might be a BC break (although I'm in favor of it)

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php:89]: Because there is a dispatchHook in this case.

File: src/Core/Grid/Definition/Factory/RequestSqlGridDefinitionFactory.php
Comment (atomiix) [inline src/Core/Grid/Definition/Factory/RequestSqlGridDefinitionFactory.php:180]: As @jolelievre stated, the constructor can be cleaned too.

File: src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php
Comment (Progi1984) [reply src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php:216]: Fixed

File: src/Core/Grid/Definition/Factory/RequestSqlGridDefinitionFactory.php
Comment (Progi1984) [reply src/Core/Grid/Definition/Factory/RequestSqlGridDefinitionFactory.php:180]: Fixed. Thanks

File: src/Core/Grid/Definition/Factory/WebserviceKeyDefinitionFactory.php
Comment (Progi1984) [reply src/Core/Grid/Definition/Factory/WebserviceKeyDefinitionFactory.php:194]: Fixed

File: src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php
Comment (Progi1984) [reply src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php]: Fixed

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/categories.yml
Comment (Progi1984) [reply src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/categories.yml]: Fixed

**Review by atomiix — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php
Comment (jolelievre) [reply src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/LogsController.php:89]: Oh I get it, so you didn't need to modify the routing either and it uses this specific search controller All good! my bad

File: src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php:56]: I don't think we need the use statement here, just for the deprecation comment
I'm even surprised CSFixer doesn't shout at this but it's probably because it takes the comment into account, anyway this is probably overkill but not worth blocking the PR either

**Review by jolelievre — DISMISSED** [review]
Only one extra `use` statement But I leave you the choice to remove it or not

**Comment by khouloudbelguith** [comment]
Hi @Progi1984  (edited)

I found an issue

1. Go to BO > Shop Parameters > SEO & Urls page
2. Try to filter by any filed
3. See error

https://drive.google.com/file/d/16KRdZ2A98DfR8b4InAiSFUePVVBwztgd/view?usp=sharing

PS: I found this regression: https://github.com/PrestaShop/PrestaShop/issues/22224 but it is reproduced without PR.
It could be fixed here?
Thanks!

**Comment by PululuK** [comment]
> PS: I found this regression: #22224 but it is reproduced without PR.
> It could be fixed here?
> Thanks!

Hi @khouloudbelguith 

This bug was introduced here : https://github.com/PrestaShop/PrestaShop/pull/21075 (I'm sorry)
Will be fixed here : https://github.com/PrestaShop/PrestaShop/pull/22229

Nice catch ! Thank you so much @khouloudbelguith

**Comment by Progi1984** [comment]
@khouloudbelguith If you test #22229 and if we merge it, I could rebase this PR after.

**Comment by khouloudbelguith** [comment]
Hi @Progi1984,

About this PR: https://github.com/PrestaShop/PrestaShop/pull/22229, it is validated.

Thanks!

**Review by atomiix — DISMISSED** [review]

**Review by matks — DISMISSED** [review]
I'm a bit not confortable with the constructor BC breaks and the services BC break 😅  but I respect the work of @jolelievre and @Progi1984 👍

**Comment by khouloudbelguith** [comment]
Hi @zuk3975, @Progi1984,

The issue of the SEO & URLs page is not fixed.

1. Go to BO > Shop Parameters > SEO & Urls page
2. Try to filter by any filed
3. See error

https://drive.google.com/file/d/1tLquW7Bb4wMNuFEEeam14gwWawPGBNEe/view?usp=sharing

Thanks!

**Comment by Progi1984** [comment]
Thanks @khouloudbelguith. It's fixed.

**Review by atomiix — DISMISSED** [review]

**Comment by Progi1984** [comment]
Thanks @matks for the label

**Review by sowbiba — DISMISSED** [review]
LGTM, unless you want to fix the comment by @jolelievre

**Review by sowbiba — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php
Comment (jolelievre) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/WebserviceController.php:62]: One think I don't get here is that this is the exact same code as in the `renderPage` So what's the point of duplicating this code instead of using the existing method?

**Comment by khouloudbelguith** [comment]
Hi @zuk3975, @Progi1984,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1u1kTevSI8bAdc1OZTXn1DGBWCEVsrGKV/view?usp=sharing

Thank you!

**Pull request merged**
